dozer-types/src/models/api_endpoint.rs

```diff
@@ -1,12 +1,74 @@
 use serde::ser::SerializeStruct;
 use serde::{Deserialize, Serialize};
+use std::fmt;
 
 #[derive(Serialize, Deserialize, Eq, PartialEq, Clone, ::prost::Message)]
 pub struct ApiIndex {
     #[prost(string, repeated, tag = "1")]
     pub primary_key: Vec<String>,
 }
 
+#[derive(Debug, Deserialize, Eq, PartialEq, Clone)]
+pub enum OnInsertResolutionTypes {
+    Nothing = 0,
+    Update = 1,
+    Panic = 2,
+}
+
+impl fmt::Display for OnInsertResolutionTypes {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            OnInsertResolutionTypes::Nothing => write!(f, "nothing"),
+            OnInsertResolutionTypes::Update => write!(f, "update"),
+            OnInsertResolutionTypes::Panic => write!(f, "panic"),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq, Clone)]
+pub enum OnUpdateResolutionTypes {
+    Nothing = 0,
+    Upsert = 1,
+    Panic = 2,
+}
+
+impl fmt::Display for OnUpdateResolutionTypes {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            OnUpdateResolutionTypes::Nothing => write!(f, "nothing"),
+            OnUpdateResolutionTypes::Upsert => write!(f, "upsert"),
+            OnUpdateResolutionTypes::Panic => write!(f, "panic"),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq, Clone)]
+pub enum OnDeleteResolutionTypes {
+    Nothing = 0,
+    Panic = 1,
+}
+
+impl fmt::Display for OnDeleteResolutionTypes {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            OnDeleteResolutionTypes::Nothing => write!(f, "nothing"),
+            OnDeleteResolutionTypes::Panic => write!(f, "panic"),
+        }
+    }
+}
+
+#[derive(Deserialize, Eq, PartialEq, Clone, ::prost::Message)]
+pub struct ConflictResolution {
+    #[prost(string, optional, tag = "1")]
```

Comment:
I think `on_insert: Option<OnInsertResolutionTypes>` will work

Reply:
Nope. `::prost::Message` trait expects value to be in int32 for enum.

Reply:
Is it possible to use something like what we did for `Connection::config`?

Reply:
Sorry just saw the implementation. This is not relavant any more.